middleware/callback.go

```diff
@@ -0,0 +1,32 @@
+package middleware
+
+import (
+	"bytes"
+	"io/ioutil"
+	"net/http"
+
+	"github.com/gol4ng/httpware"
+)
+
+func Callback(callbackFunc func(*ResponseWriterInterceptor, *http.Request)) httpware.Middleware {
+	return func(next http.Handler) http.Handler {
+		return http.HandlerFunc(func(writer http.ResponseWriter, req *http.Request) {
+			buf := new(bytes.Buffer)
+			buf.ReadFrom(req.Body)
+			// reset io.Reader because is already read
+			req.Body = ioutil.NopCloser(bytes.NewBuffer(buf.Bytes()))
```

Comment:
hmmm I'm not sure that thing is good

Reply:
https://medium.com/@xoen/golang-read-from-an-io-readwriter-without-loosing-its-content-2c6911805361#.2j3x7l5cm

OK good 👍